errors.go

```diff
@@ -109,6 +124,52 @@ func sprintf(format string, args ...interface{}) string {
 	return fmt.Sprintf(format, args...)
 }
 
+// NewBatch creates a new BatchError.
+func NewBatch(errs []error) BatchError {
+	return &batcherrtype{
+		errors: errs,
+	}
+}
+
+//Append appends the error
+func (b *batcherrtype) Append(err error) {
+	b.errors = append(b.errors, err)
+}
+
+//Errors returns the errors
+func (b *batcherrtype) Errors() []error {
+	return b.errors
+}
+
+//IsEmpty returns if there is any error
+func (b *batcherrtype) IsEmpty() bool {
```

Comment:
`IsEmpty` is not necessary anymore. Since we agree that the presence of a BatchError means there _is_ an error (not nil).